### DISCUSSION TITLE
Rename tag for latest with prefix/suffix

### DISCUSSION THREAD

**Discussion by rigon** [docker/metadata-action#468] [Q&A]
I have a project where I build two images:
- the first has the usual tags i.e. follow semver and latest is updated every time there is a new version;
- the second is tagged as the first, but with the suffix `-slim`

For that, I have the following configuration:

```yaml
- name: Metadata
  uses: docker/metadata-action@v5
  with:
        flavor: |
          latest=auto
          suffix=-slim,onlatest=true
        tags: |
          type=ref,event=branch
          type=ref,event=pr
          type=semver,pattern={{version}}
          type=semver,pattern={{major}}.{{minor}}
          type=semver,pattern={{major}}
```

However, for the latest tag on the slim version, I’m getting `latest-slim`. Instead, I would like this tag to be simply `slim` for the latest version of the slim image.

How can I adjust my configuration to achieve this?

**Comment by polarathene** [comment]
> Instead, I would like this tag to be simply `slim` for the latest version of the slim image.
> 
> How can I adjust my configuration to achieve this?

`flavor.latest=auto` generates a `latest` tag when your `type=semver` tag rules are processed.

`suffix=-slim` appends that suffix to each of those semver produced tags, and the `onlatest=true` appends it to `latest`, thus `latest-slim`.

I've [documented this information](https://github.com/docker/metadata-action/issues/593) if you'd like to understand that part of the action better.

---

I assume the snippet you shared is only meant to represent your `-slim` variant, so you can just change flavor to:

```yaml
flavor: |
  latest=false
  suffix=-slim
```

Then manually append or prepend a `${{ env.image_name }}:slim` tag to use with your tags output.

I can't recall if you can opt-out of a suffix, so the other approach would be:

```yaml
flavor: |
  latest=false
tags: |
  type=ref,event=branch,suffix=-slim
  type=ref,event=pr,suffix=-slim
  type=semver,pattern={{version}},suffix=-slim
  type=semver,pattern={{major}}.{{minor}},suffix=-slim
  type=semver,pattern={{major}},suffix=-slim
```

You could also take the non-slim tags and process them with either Docker Bake HCL (_and use the equivalent `docker/bake-action` instead of `docker/build-push-action`_) or through a step that runs a script to add your changes:

```yaml
- name: Create image tags for the slim variant
  id: image-tags-slim
  uses: actions/github-script@v8
  env:
    JSON_IMAGE_METADATA: ${{ steps.image-metadata.outputs.json }}
    UNTAGGED_IMAGE_REF: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
  with:
    script: |
      const tags = JSON.parse(process.env.JSON_IMAGE_METADATA).tags
      const add_tag = (tag) => `${process.env.UNTAGGED_IMAGE_REF}:${ tag }` 

      const variant = 'slim'
      const tag_list = tags.map(tag => {
        if tag.endsWith(':latest') {
          return add_tag(`${ variant }`)
        } else {
          return `${ tag }-${ variant }`
        }
      })

      await core.group(`Prepared ${ tag_list.length } tags`, async () => {
        tag_list.forEach(tag => core.info(tag))
      })

      return tag_list.join(',')
```

Then you can use `${{ steps.image-tags-slim.outputs.result }}` for your slim image publish, and `${{ steps.image-metadata.outputs.tags }}` is your main image variant that I assume you use with a separate `docker/build-push-action`?.
